helpers.go

```diff
@@ -332,28 +332,13 @@ var getBytesImmutable = func(s string) (b []byte) {
 
 // HTTP methods and their unique INTs
 func methodInt(s string) int {
-	switch s {
-	case MethodGet:
-		return 0
-	case MethodHead:
-		return 1
-	case MethodPost:
-		return 2
-	case MethodPut:
-		return 3
-	case MethodDelete:
-		return 4
-	case MethodConnect:
-		return 5
-	case MethodOptions:
-		return 6
-	case MethodTrace:
-		return 7
-	case MethodPatch:
-		return 8
-	default:
-		return -1
+	for i, v := range intMethod {
```

Comment:
did you check the performance, we had added the switch at that time because the access is faster there than with the map
can you make a comparison here

we should keep the functions which are used for the main path of the application as fast as possible

Reply:
not quite sure about the performance, but I think I need this implementation for the custom methods. If we using switch case statement it will only work for static methods but for this feature it will be dynamic.

Reply:
Right, maybe we should switch back to a map as well

can you test if the slice-for or the map access is faster by creating a benchmark and posting metrics

Reply:
this is the benchmark for forloop implementation and switch-case statement:

- For Loop

```
goos: darwin
goarch: amd64
pkg: github.com/gofiber/fiber/v2
cpu: Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
Benchmark_MethodInt
Benchmark_MethodInt-4   	55310788	        20.03 ns/op
PASS
```

- Switch-Case Statement

```
goos: darwin
goarch: amd64
pkg: github.com/gofiber/fiber/v2
cpu: Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
Benchmark_MethodInt
Benchmark_MethodInt-4   	1000000000	         0.3897 ns/op
PASS
```

Switch case is far way better, I think we can keep the switch case and handle the dynamic route on default statement, it will be like this:
```go
func methodInt(s string) int {
	switch s {
	case MethodGet:
		return 0
	case MethodHead:
		return 1
	case MethodPost:
		return 2
	case MethodPut:
		return 3
	case MethodDelete:
		return 4
	case MethodConnect:
		return 5
	case MethodOptions:
		return 6
	case MethodTrace:
		return 7
	case MethodPatch:
		return 8
	default:
		for i, v := range intMethod[8:] {
			if s == v {
				return i
			}
		}
		return -1
	}
}
```

I did the benchmark for this and this is the result:
- for method that hardcoded:
```
goos: darwin
goarch: amd64
pkg: github.com/gofiber/fiber/v2
cpu: Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
Benchmark_MethodInt
Benchmark_MethodInt-4   	1000000000	         0.3713 ns/op
PASS
```
- for dynamic method
```
goos: darwin
goarch: amd64
pkg: github.com/gofiber/fiber/v2
cpu: Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
Benchmark_MethodInt
Benchmark_MethodInt-4   	244620964	         4.902 ns/op
PASS
```

with this approach, the performance of the hardcoded method will be the same and the performance of dynamic method will be better from the first approach although it wont be better than the hardcoded method. What do you think?

Reply:
this is okay for me